text/0000-deprecate-anonymous-parameters.md

```diff
@@ -0,0 +1,205 @@
+- Feature Name: deprecate_anonymous_parameters
+- Start Date: 2016-07-19
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Currently Rust allows anonymous parameters in trait methods:
+
+```Rust
+trait T {
+    fn foo(i32);
+
+    fn bar_with_default_impl(String, String) {
+
+    }
+}
+```
+
+This RFC proposes to deprecate this syntax. This RFC intentionally does not
+propose to remove this syntax.
+
+# Motivation
+[motivation]: #motivation
+
+Anonymous parameters are a [historic accident]. They cause a number of technical
```

Comment:
This claim is misrepresenting the text it references.

What Niko wrote there was:
> I don't believe it was intended that parameter names can be omitted in trait methods *with* a body

(emphasis added)

I do not think it was *ever* an accident that the design allowed anonymous parameters for trait methods *without* a body.